### PR TITLE
[Classical Shadow 1] Minor Updates to the Quantum Processing Functions to adjust for robust shadow.

### PR DESCRIPTION
**Pull request by Min-Li**
Minor Updates to the Quantum Processing Functions to adjust for robust shadow.

Add optional qubits argument to shadow workflow.

---

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: mitiq/shadows/quantum_processing.py
Comment (natestemen) [inline mitiq/shadows/quantum_processing.py:110]: Here's my suggestion. I didn't want to use the suggestion tool since I cannot properly format the code here. I'll leave that to you.

> Tuple containing two lists of strings, each of length equal to `n_total_measurements`. Strings in the first list are sequences of 0's and 1's which represent qubit measurements in the computational basis (e.g. "01001"). Strings in the second list are sequences of Pauli operators which were performed on the circuit (e.g. "XZZII").

File: mitiq/shadows/test/test_quantum_processing.py
Comment (natestemen) [inline mitiq/shadows/test/test_quantum_processing.py:163]: TIL you can assign a variable as a tuple with surrounding the expression in parens! Cool 😎.

**Review by natestemen — COMMENTED** [review]
Nice well contained PR. Looks good, just a few minor things.

For next time, try to use more descriptive PR titles. For this one something like "Add optional `qubits` argument to shadow workflow". Specificity is helpful!

File: mitiq/shadows/quantum_processing.py
Comment (natestemen) [inline mitiq/shadows/quantum_processing.py:101]: I think we want the double backticks around `executor` and `MeasurementResult`. With just one it renders as italic, not code.
Screenshot 2023-08-02 at 7 41 50 AM

File: mitiq/shadows/quantum_processing.py
Comment (Misty-W) [inline mitiq/shadows/quantum_processing.py:54]:
```suggestion
        qubits: The qubits to measure. If None, all qubits in the circuit.
```

**Review by Misty-W — COMMENTED** [review]
Nice work, @Min-Li! This PR can be merged after Nate's comment on the docstring is addressed.

**Review by andreamari — APPROVED** [review]
Besides Nate's comments, LGTM!

File: mitiq/shadows/quantum_processing.py
Comment (Misty-W) [inline mitiq/shadows/quantum_processing.py:101]:
```suggestion
        The ``executor`` must return a ``MeasurementResult`` for a single shot,
        i.e., a single bitstring.
```

File: mitiq/shadows/quantum_processing.py
Comment (Misty-W) [inline mitiq/shadows/quantum_processing.py:113]:
```suggestion
        Tuple containing two lists of strings, each of length equal to
        `n_total_measurements`. Strings in the first list are sequences of 0's and
        1's, which represent qubit measurements in the computational basis (e.g.
        "01001"). Strings in the second list are sequences of Pauli operators
        performed on the circuit (e.g. "XZZII").
```

File: mitiq/shadows/quantum_processing.py
Comment (Misty-W) [reply mitiq/shadows/quantum_processing.py:101]: Added suggestion to this effect.

File: mitiq/shadows/quantum_processing.py
Comment (Misty-W) [reply mitiq/shadows/quantum_processing.py]: Added suggestion with slightly abbreviated wording.

File: mitiq/shadows/quantum_processing.py
Comment (Min-Li) [reply mitiq/shadows/quantum_processing.py]: Thanks for your suggestion Misty, Please note that the change of the doc has a minor mistake, the string in the second list just represents the Pauli-basis measurement performed on each qubit, not the "sequences of Pauli operators performed" (I think you mean rotation unitaries, but actually it's not), that's why it can't be "I". Just a minor suggestion, please Let me know before you modify it!

**Pull request merged**
